exporter/exporterhelper/queued_retry.go

```diff
@@ -79,21 +81,42 @@ type queuedRetrySender struct {
 	consumerSender requestSender
 	queue          *queue.BoundedQueue
 	retryStopCh    chan struct{}
+	logger         *zap.Logger
 }
 
-var errorRefused = errors.New("failed to add to the queue")
+func createSampledLogger(logger *zap.Logger) *zap.Logger {
+	if logger.Core().Enabled(zapcore.DebugLevel) {
+		// Debugging is enabled. Don't do any sampling.
+		return logger
+	}
+
+	// Create a logger that samples all messages to 1 per 10 seconds initially,
+	// and 1/100 of messages after that.
+	opts := zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+		return zapcore.NewSamplerWithOptions(
+			core,
+			10*time.Second,
+			1,
+			100,
```

Comment:
Is it possible these values will need to be configurable based on the volume of the deployment?

Reply:
Possibly, we can add if needed.

The queued_retry does fairly low and predictable amount of logging of which it is virtually always sufficient to just see the first failure since the most common case is exporter failing because of misconfiguration, server being down or some transient problem with the server. In all of these cases there is no need for full forensic analysis using logs, the fist log record is typically sufficient and heavy sampling to 1% is good too. Since the volume is low I don't expect log flooding from this and neither I expect to have a need to increase the sampling to see more logs.

Reply:
@jrcamp ok with this?